src/components/content/order/common/FlavorPrice.tsx

```diff
@@ -3,17 +3,51 @@
  * SPDX-FileCopyrightText: Huawei Inc.
  */
 
-import { Tag } from 'antd';
+import { Tag, Typography } from 'antd';
 import React from 'react';
+import flavorStyles from '../../../../styles/flavor.module.css';
 import serviceModifyStyles from '../../../../styles/service-modify.module.css';
+import { ServiceFlavor } from '../../../../xpanse-api/generated';
+import { ServiceFlavorWithPriceResult } from '../types/ServiceFlavorWithPrice.ts';
 
-export const FlavorPrice = (): React.JSX.Element => {
+export const FlavorPrice = (
+    flavor: ServiceFlavor,
+    serviceFlavorWithPriceList: ServiceFlavorWithPriceResult[]
+): React.JSX.Element => {
+    const { Text } = Typography;
+    const convertToFlavorMap = (
+        serviceFlavorWithPriceList: ServiceFlavorWithPriceResult[]
+    ): Record<string, ServiceFlavorWithPriceResult> => {
+        return serviceFlavorWithPriceList.reduce<Record<string, ServiceFlavorWithPriceResult>>((map, flavor) => {
+            map[flavor.name] = flavor;
+            return map;
+        }, {});
+    };
+    const flavorMap = convertToFlavorMap(serviceFlavorWithPriceList);
     return (
         <>
-            <Tag color={'blue'} className={serviceModifyStyles.flavorPriceContent}>
-                {/* TODO Will be fixed after #1597 is fixed */}
-                {(20).toString().concat(' ').concat('EUR/').concat('hour')}
-            </Tag>
+            {flavor.name && flavorMap[flavor.name].price.recurringPrice ? (
+                <Tag color={'blue'} className={serviceModifyStyles.flavorPriceContent}>
+                    {flavorMap[flavor.name].price.recurringPrice?.cost
+                        .toString()
+                        .concat(' ')
+                        .concat(flavorMap[flavor.name].price.recurringPrice?.currency ?? '')
+                        .concat('/')
+                        .concat(flavorMap[flavor.name].price.recurringPrice?.period ?? '')}
+                </Tag>
+            ) : null}
+            {flavorMap[flavor.name].price.oneTimePaymentPrice ? (
+                <div className={flavorStyles.flavorOneTimePriceContent}>
+                    <Text type={'secondary'} className={flavorStyles.antTypography}>
+                        {flavorMap[flavor.name].price.oneTimePaymentPrice?.cost
+                            .toString()
+                            .concat(' ')
+                            .concat(flavorMap[flavor.name].price.oneTimePaymentPrice?.currency ?? '')
+                            .concat('/')
+                            .concat(flavorMap[flavor.name].price.oneTimePaymentPrice?.period ?? '')}
```

Comment:
i think we need a mapping here for period
yearly -> year
montly -> month
daily -> day
hourly -> hour
oneTime -> one time